pkg/devfile/parser/parse.go

```diff
@@ -202,13 +232,21 @@ func parseParentAndPlugin(d DevfileObj) (err error) {
 		if component.Plugin != nil && !reflect.DeepEqual(component.Plugin, &v1.PluginComponent{}) {
 			plugin := component.Plugin
 			var pluginDevfileObj DevfileObj
-			if plugin.Uri != "" {
-				pluginDevfileObj, err = parseFromURI(plugin.Uri, d.Ctx)
+			switch {
+			case plugin.Uri != "":
+				pluginDevfileObj, err = parseFromURI(plugin.ImportReference, d.Ctx, resolveCtx, tool)
 				if err != nil {
 					return err
 				}
-			} else {
-				return fmt.Errorf("plugin URI undefined, currently only URI is suppported")
+			case plugin.Id != "":
+				pluginDevfileObj, err = parseFromRegistry(plugin.ImportReference, resolveCtx, tool)
+				if err != nil {
+					return err
+				}
+			case plugin.Kubernetes != nil:
+				pluginDevfileObj, err = parseFromKubeCRD(plugin.ImportReference, resolveCtx, tool)
+			default:
+				return fmt.Errorf("plugin URI or plugin Id undefined, currently only URI and Id are suppported")
```

Comment:
We want to do this because plugin is only removed from schema, still left in spec.  Kubernetes CRD still can contain plugins, and parsing from that crd also need to parse plugins to get all info flattened. I have confirmed that with Elson in the meeting